shared/modules/types.go

```diff
@@ -1,6 +1,228 @@
-package types
+package modules
 
-// TODO(pocket/issues/163): Move this into utility
+import (
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// This file contains the minimum shared structures (GenesisState) and the many shared interfaces the modules implement
+// the main purpose of this structure is to ensure the ownership of the
+
+type GenesisState struct {
+	PersistenceGenesisState PersistenceGenesisState `json:"persistenceGenesisState"`
+	ConsensusGenesisState   ConsensusGenesisState   `json:"consensusGenesisState"`
+}
+
+type BaseConfig struct {
+	RootDirectory string `json:"root_directory"`
+	PrivateKey    string `json:"private_key"` // TODO (team) better architecture for key management (keybase, keyfiles, etc.)
+}
+
+type Config struct {
+	Base        *BaseConfig       `json:"base"`
+	Consensus   ConsensusConfig   `json:"consensus"`
+	Utility     UtilityConfig     `json:"utility"`
+	Persistence PersistenceConfig `json:"persistence"`
+	P2P         P2PConfig         `json:"p2p"`
+	Telemetry   TelemetryConfig   `json:"telemetry"`
+}
+
+type ConsensusConfig interface {
+	GetMaxMempoolBytes() uint64
+	GetPaceMakerConfig() PacemakerConfig
+}
+
+type PacemakerConfig interface {
+	SetTimeoutMsec(uint64)
+	GetTimeoutMsec() uint64
+	GetManual() bool
+	GetDebugTimeBetweenStepsMsec() uint64
+}
+
+type PersistenceConfig interface {
+	GetPostgresUrl() string
+	GetNodeSchema() string
+	GetBlockStorePath() string
+}
+
+type P2PConfig interface {
+	GetConsensusPort() uint32
+	GetUseRainTree() bool
+	IsEmptyConnType() bool // TODO (team) make enum
+}
+
+type TelemetryConfig interface {
+	GetEnabled() bool
+	GetAddress() string
+	GetEndpoint() string
+}
+
+type UtilityConfig interface{}
```

Comment:
Since we're creating an empty placeholder `UtilityConfig`, should we do the same for the genesis state for P2P and utility too?

Reply:
Mainly I was just going for conssitency, but if I had to think of examples:

`p2p genesis state` - initial seeders for the addrbook
`utility genesis state` - initial servicers and the DAO permissioned fisherman

Reply:
Peers / seeds are not part of the `state` so this seems counter intuitive to me.

> initial servicers and the DAO permissioned fisherman

^ Is already in the persistence genesis state